### ISSUE TITLE
Size-1 crop axis: resample/functional 2D crops return NaN, functional 3D crops raise _LinAlgError

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4747]
## Summary

A crop with a size-1 axis is still broken everywhere the crop matrix is solved from the box vertices with `get_perspective_transform` / `get_perspective_transform3d`. A size-1 axis makes the vertices collinear (2D) or coplanar (3D). In 2D the solve returns **NaN silently**. In 3D it raises `_LinAlgError`. #4732 fixes only the two 3D augmentation classes (#4705). It builds their matrix as the translation it is, and it leaves these paths as they are:

```python
import torch, kornia.augmentation as K
from kornia.geometry.transform import center_crop, crop_by_boxes, crop_and_resize, center_crop3d, crop_and_resize3d

x = torch.arange(20.).reshape(1, 1, 4, 5)
K.CenterCrop((1, 3), p=1.0, cropping_mode="resample")(x)   # tensor([[[[nan, nan, nan]]]])
K.RandomCrop((2, 1), p=1.0, cropping_mode="resample")(x)   # [nan, nan]
center_crop(x, (1, 3))                                     # [nan, nan, nan]
src = torch.tensor([[[1., 1.], [3., 1.], [3., 1.], [1., 1.]]])
dst = torch.tensor([[[0., 0.], [2., 0.], [2., 0.], [0., 0.]]])
crop_by_boxes(x, src, dst)                                 # [nan, nan, nan]
crop_and_resize(x, src, (1, 3))                            # [nan, nan, nan]

v = torch.rand(1, 1, 4, 5, 6)
center_crop3d(v, (1, 3, 4))                                # _LinAlgError: ... input matrix is singular
crop_and_resize3d(v, <box with a size-1 depth>, (1, 3, 4)) # _LinAlgError
```

The default `cropping_mode="slice"` of `CenterCrop` / `RandomCrop` is correct (`[6., 7., 8.]` above). The NaN appears only on the resample path and in the functionals. Measured on `main` @ 8b03a8085, torch 2.14.0, CPU float32.

## Expected

Each call returns the single row or slice, the same as the slice path. When the source and destination boxes have the same extent, the crop is a pure translation and needs no solve; #4732's `_crop_translation3d` is that construction for 3D. `crop_and_resize*` can also scale, so a degenerate axis there needs its scale taken from the box extents rather than from a DLT solve. If a size-1 axis is not going to be supported, it should raise a `ValueError` naming the constraint. A silent NaN is the worst outcome of the three.

The functional 3D crops live in `kornia/geometry/transform/crop3d.py`, which #4556 is changing. The author of #4732 left them out for that reason. The 2D NaN was first reported in #4732's description.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
